py-publishing.Rmd

```diff
@@ -0,0 +1,487 @@
+for example https://merely-useful.github.io/py-publishing.html,
+the following things happen:
+
+1. The URL `https://merely-useful.github.io` points to a location on another computer
+   where the relevant files for this website reside.
+   This computer is known as the website host or server,
+   since it *hosts* the webpage files on its local storage drives
+   and *serves* them to us via the internet.
+2. The browser requests the file `py-publishing.html` from the host.
+3. The browser reads the contents of `py-publishing.html` and displays it for you in the browser window.
+
+In reverse order this process also describes what you need to do to get your own website online:
+
+1. You need an HTML file that describes what people should see on your page.
+2. You need to host the HTML file on a computer on the internet.
+3. You need a way to associate a URL with the address of your host.
+
+In this chapter,
+you'll learn a process for getting your work online that leverages what you already know ---
+creating HTML files using R Markdown (#1 above) and how to host your work in a GitHub repository (#2).
+The third step will be handled by a GitHub service called [GitHub Pages][gh-pages].
+By following the conventions that GitHub Pages expects,
+you'll be able to make a webpage for any of your repositories available at:
+`http://{{your_username}}.github.io/{{repo_name}}`.
+
+## How do I get my work on the web?
+
+### A starting point
+
+TODO: revisit this later when more content is fleshed out.  Maybe there will be a repo we can rely on all learners having that we can start from.
+
+In practice you'll probably start thinking about a website once you've already done a lot of work on a project ---
+your project will already have some analysis documented in R Markdown,
+be in version control and hosted on GitHub.
+However,
+so that we can work with a specific example,
+you'll set up a project in this section that is less developed
+than where your project might be when you start thinking about making a website.
+
+At the end of this section,
+you should have a project folder with an example report in `report.Rmd`.
+This project should also be on GitHub at `https://github.com/{{your_username}}/sharing-work`,
+where `{{your_username}}` should be substituted with your GitHub username,
+e.g. mine is at https://github.com/cwickham/sharing-work.
+
+Let's start by creating a GitHub repository called `sharing-work`,
+by following illustrated steps in [the GitHub guide][github-new-repo].
+
+[github-new-repo]: https://help.github.com/en/articles/create-a-repo
+
+Open JupyterLab and create a new Jupyter Notebook
+("File -> New -> Notebook", or the Python symbol on the launcher page)
+Rename this file to `report.ipynb` and pair it with an `.Rmd` file using Jupytext.
+The pairing to an `.Rmd` file is not necessary,
+but enables a more human-friendly version control workflow.
+
+In the first cell create a markdown title that reads "Sharing work".
```

Comment:
You've written this like learners know what a "markdown title" is...do they?

Reply:
Yes, this is included in the Jupyter section I just made, but I reworded to clarify here also.